cmake/modules/AddSwift.cmake

```diff
@@ -145,26 +147,15 @@ function(_add_host_variant_c_compile_flags target)
         endif()
       endif()
     endif()
```

Comment:
Is there a reason to not just put `-momit-leaf-frame-pointer`/`/Oy` into `CMAKE_CXX_FLAGS_RELEASE`?

At the top level CMakeLists, we can do something like:
```
if(NOT LLVM_ENABLE_ASSERTIONS)
  if(SWIFT_COMPILER_IS_MSVC_LIKE)
    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /Oy")
  else()
    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -momit-leaf-frame-pointer")
  endif()
endif()
```

Reply:
The current logic only applies -momit-leaf-frame-pointer for i386/i686 architectures. I think that means it cannot be folded into CMAKE_CXX_FLAGS_RELEASE because that one applies to all architectures?

Reply:
We should be able to add that to the condition check right?

Reply:
Added a comment explaining what this can't be done: Darwin builds do multiple SDKs in one build, and they have different architectures.